### ISSUE TITLE
`ServerConfig::set_single_cert` unwraps private key in `handy::AlwaysResolvesChain::new_rsa`

### ISSUE THREAD

**Issue by sp3d** [ctz/rustls#156]
It would be sufficient to allow applications to handle errors in set_single_cert.
More generally it would be nice if the PrivateKey type had the invariant of actually holding a valid private key rather than arbitrary bytes.

I don't know enough to say whether the latter is feasible to implement (e.g. if there are other factors on which the validity of a private key depends which aren't known when PrivateKey values are created).

One option is that rather than panicking this function could simply returned a `Result`, possibly taking `self` by value so that its return value could not be ignored.

This manifested in a program of mine with the following excerpted backtrace, using rustls 0.12.0:

```
thread 'main' panicked at 'Invalid RSA private key: ()', libcore/result.rs:945:5
stack backtrace:
...
   7: core::panicking::panic_fmt
             at libcore/panicking.rs:71
   8: core::result::unwrap_failed
   9: rustls::server::handy::AlwaysResolvesChain::new_rsa
  10: rustls::server::ServerConfig::set_single_cert
...
```

Thanks for your time and a very useful library.

**Comment by ctz** [comment]
> One option is that rather than panicking this function could simply returned a Result

Yes. That it doesn't do this is a mistake.

**Comment by ctz** [comment]
Corrected in 5fa72777.
